packages/contract/src/contract-factory.ts

```diff
@@ -372,6 +376,77 @@ export default class ContractFactory {
     return { waitForResult, contractId, waitForTransactionId };
   }
 
+  async deployAsBlobTxForScript(configurableConstants: { [name: string]: unknown } = {}): Promise<{
```

Comment:
Why did we choose to add this logic on the contract factory and not on the script class?

Reply:
Were you thinking a script-factory or just on the script class itself?